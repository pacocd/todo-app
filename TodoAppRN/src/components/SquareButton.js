import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';

const SquareButton = ({ color, textColor, text, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress} style={{ margin: 16 }}>
      <View
        style={{
          borderRadius: 5,
          backgroundColor: color,
          height: 50,
          flex: 1,
          justifyContent: 'center'
        }}
      >
        <Text style={{ color: textColor, alignSelf: 'center' }}>{text}</Text>
      </View>
    </TouchableOpacity>
  );
};

SquareButton.propTypes = {
  color: PropTypes.string.isRequired,
  textColor: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired
};

export default SquareButton;
