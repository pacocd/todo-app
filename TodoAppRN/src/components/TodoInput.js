import React, { Component } from 'react';
import { View, Text, TextInput, Switch } from 'react-native';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { setEditingFieldsData } from '../redux/actions';

export class TodoInput extends Component {
  static propTypes = {
    fieldName: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.bool,
      PropTypes.number
    ]),
    type: PropTypes.string.isRequired,
    setEditingFieldsData: PropTypes.func.isRequired
  };

  state = {
    value: this.props.value
  };

  componentDidMount() {
    const { fieldName, value } = this.props;

    this.props.setEditingFieldsData(fieldName, value);
  }

  handleOnChange = value => {
    const { fieldName } = this.props;
    this.props.setEditingFieldsData(fieldName, value);
    this.setState({ value });
  };

  getGenericInput = (keyboardType, title, value) => {
    return (
      <View>
        <Text style={{ marginBottom: 10, fontWeight: 'bold', fontSize: 12 }}>
          {title}
        </Text>
        <TextInput
          placeholder="Write here"
          value={`${value}`}
          keyboardType={keyboardType}
          onChangeText={this.handleOnChange}
          underlineColorAndroid="white"
          multiline
        />
      </View>
    );
  };

  getSwitch = (title, value) => {
    return (
      <View style={{ flexDirection: 'row', justifyContent: 'space-between' }}>
        <Text style={{ marginBottom: 10, fontWeight: 'bold', fontSize: 12 }}>
          {title}
        </Text>
        <Switch value={value} onValueChange={this.handleOnChange} />
      </View>
    );
  };

  renderInput(type, title, value) {
    switch (type) {
      case 'switch':
        return this.getSwitch(title, value);
      case 'numeric':
        return this.getGenericInput('numeric', title, value);
      default:
        return this.getGenericInput('default', title, value);
    }
  }

  render() {
    const { title, type } = this.props;
    const { value } = this.state;
    return (
      <View style={{ margin: 16 }}>{this.renderInput(type, title, value)}</View>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    setEditingFieldsData: (fieldName, value) =>
      dispatch(setEditingFieldsData(fieldName, value))
  };
};

export default connect(
  undefined,
  mapDispatchToProps
)(TodoInput);
