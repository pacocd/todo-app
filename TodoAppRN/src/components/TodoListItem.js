import React, { Component } from 'react';
import { Text, View, Switch } from 'react-native';
import PropTypes from 'prop-types';

export class TodoListItem extends Component {
  static propTypes = {
    todo: PropTypes.shape({
      id: PropTypes.number.isRequired,
      user_id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      completed: PropTypes.bool.isRequired
    }).isRequired
  };

  render() {
    const { title } = this.props.todo;
    return (
      <View style={{ width: '100%', paddingVertical: 14 }}>
        <Text>{title}</Text>
      </View>
    );
  }
}

export default TodoListItem;
