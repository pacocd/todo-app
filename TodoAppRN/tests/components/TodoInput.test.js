import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { TodoInput } from '../../src/components/TodoInput';
configure({ adapter: new Adapter() });

describe('when TodoInput is fully rendered using Switch mode', () => {
  const props = {
    fieldName: 'completed',
    title: 'Completed',
    value: false,
    type: 'switch',
    setEditingFieldsData: jest.fn()
  };

  const wrapper = shallow(<TodoInput {...props} />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('when TodoInput is fully rendered using Numeric mode', () => {
  const props = {
    fieldName: 'user_id',
    title: 'User',
    value: 1,
    type: 'numeric',
    setEditingFieldsData: jest.fn()
  };

  const wrapper = shallow(<TodoInput {...props} />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('when TodoInput is fully rendered using default mode', () => {
  const props = {
    fieldName: 'title',
    title: 'Title',
    value: 'test',
    type: 'default',
    setEditingFieldsData: jest.fn()
  };

  const wrapper = shallow(<TodoInput {...props} />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
