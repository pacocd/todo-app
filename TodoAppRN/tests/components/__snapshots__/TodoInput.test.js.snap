// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when TodoInput is fully rendered using Numeric mode renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoInput
    fieldName="user_id"
    setEditingFieldsData={
      [MockFunction] {
        "calls": Array [
          Array [
            "user_id",
            1,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    title="User"
    type="numeric"
    value={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Component>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "marginBottom": 10,
            }
          }
        >
          User
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="numeric"
          multiline={true}
          onChangeText={[Function]}
          placeholder="Write here"
          underlineColorAndroid="white"
          value="1"
        />
      </Component>,
      "style": Object {
        "margin": 16,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 10,
              }
            }
          >
            User
          </Text>,
          <TextInput
            allowFontScaling={true}
            keyboardType="numeric"
            multiline={true}
            onChangeText={[Function]}
            placeholder="Write here"
            underlineColorAndroid="white"
            value="1"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "User",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "marginBottom": 10,
            },
          },
          "ref": null,
          "rendered": "User",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "keyboardType": "numeric",
            "multiline": true,
            "onChangeText": [Function],
            "placeholder": "Write here",
            "underlineColorAndroid": "white",
            "value": "1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 10,
              }
            }
          >
            User
          </Text>
          <TextInput
            allowFontScaling={true}
            keyboardType="numeric"
            multiline={true}
            onChangeText={[Function]}
            placeholder="Write here"
            underlineColorAndroid="white"
            value="1"
          />
        </Component>,
        "style": Object {
          "margin": 16,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "marginBottom": 10,
                }
              }
            >
              User
            </Text>,
            <TextInput
              allowFontScaling={true}
              keyboardType="numeric"
              multiline={true}
              onChangeText={[Function]}
              placeholder="Write here"
              underlineColorAndroid="white"
              value="1"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "User",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 10,
              },
            },
            "ref": null,
            "rendered": "User",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "keyboardType": "numeric",
              "multiline": true,
              "onChangeText": [Function],
              "placeholder": "Write here",
              "underlineColorAndroid": "white",
              "value": "1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`when TodoInput is fully rendered using Switch mode renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoInput
    fieldName="completed"
    setEditingFieldsData={
      [MockFunction] {
        "calls": Array [
          Array [
            "completed",
            false,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    title="Completed"
    type="switch"
    value={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Component
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "marginBottom": 10,
            }
          }
        >
          Completed
        </Text>
        <Switch
          disabled={false}
          onValueChange={[Function]}
          value={false}
        />
      </Component>,
      "style": Object {
        "margin": 16,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 10,
              }
            }
          >
            Completed
          </Text>,
          <Switch
            disabled={false}
            onValueChange={[Function]}
            value={false}
          />,
        ],
        "style": Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Completed",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "marginBottom": 10,
            },
          },
          "ref": null,
          "rendered": "Completed",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "onValueChange": [Function],
            "value": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 10,
              }
            }
          >
            Completed
          </Text>
          <Switch
            disabled={false}
            onValueChange={[Function]}
            value={false}
          />
        </Component>,
        "style": Object {
          "margin": 16,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "marginBottom": 10,
                }
              }
            >
              Completed
            </Text>,
            <Switch
              disabled={false}
              onValueChange={[Function]}
              value={false}
            />,
          ],
          "style": Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Completed",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 10,
              },
            },
            "ref": null,
            "rendered": "Completed",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "onValueChange": [Function],
              "value": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`when TodoInput is fully rendered using default mode renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoInput
    fieldName="title"
    setEditingFieldsData={
      [MockFunction] {
        "calls": Array [
          Array [
            "title",
            "test",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    title="Title"
    type="default"
    value="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Component>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "marginBottom": 10,
            }
          }
        >
          Title
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="default"
          multiline={true}
          onChangeText={[Function]}
          placeholder="Write here"
          underlineColorAndroid="white"
          value="test"
        />
      </Component>,
      "style": Object {
        "margin": 16,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 10,
              }
            }
          >
            Title
          </Text>,
          <TextInput
            allowFontScaling={true}
            keyboardType="default"
            multiline={true}
            onChangeText={[Function]}
            placeholder="Write here"
            underlineColorAndroid="white"
            value="test"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Title",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "marginBottom": 10,
            },
          },
          "ref": null,
          "rendered": "Title",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "keyboardType": "default",
            "multiline": true,
            "onChangeText": [Function],
            "placeholder": "Write here",
            "underlineColorAndroid": "white",
            "value": "test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 10,
              }
            }
          >
            Title
          </Text>
          <TextInput
            allowFontScaling={true}
            keyboardType="default"
            multiline={true}
            onChangeText={[Function]}
            placeholder="Write here"
            underlineColorAndroid="white"
            value="test"
          />
        </Component>,
        "style": Object {
          "margin": 16,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "marginBottom": 10,
                }
              }
            >
              Title
            </Text>,
            <TextInput
              allowFontScaling={true}
              keyboardType="default"
              multiline={true}
              onChangeText={[Function]}
              placeholder="Write here"
              underlineColorAndroid="white"
              value="test"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Title",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 10,
              },
            },
            "ref": null,
            "rendered": "Title",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "keyboardType": "default",
              "multiline": true,
              "onChangeText": [Function],
              "placeholder": "Write here",
              "underlineColorAndroid": "white",
              "value": "test",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
